cognite/client/_api/events.py

```diff
@@ -443,3 +443,38 @@ def search(self, description: str = None, filter: Union[EventFilter, Dict] = Non
                 >>> res = c.events.search(description="some description")
         """
         return self._search(list_cls=EventList, search={"description": description}, filter=filter or {}, limit=limit)
+
+    @overload
+    def upsert(self, item: Sequence[Event]) -> EventList:
+        ...
+
+    @overload
+    def upsert(self, item: Event) -> Event:
+        ...
+
+    def upsert(self, item: Event | Sequence[Event]) -> Event | EventList:
```

Comment:
Since this method doesn't support `EventUpdate`s, the upsert mode is a weird mix between overwrite and "leave as is":

```py
>>> ev = Event(external_id="x", metadata={"a": "1"}, type="foo")
>>> client.events.upsert(ev)
>>> ev.type = None
>>> ev.metadata = {"b": "2"}
>>> client.events.upsert(ev).dump()
{'type': 'foo',
 'metadata': {'b': '2'},
 ...}
```
Here, the `a` key in `metadata` is overwritten, but the event `type` is left as is. There would be no way for the user of the `upsert` method to remove an attribute i.e. set it to null.

I'm not sure what I think is best, but I think either the mode should be "overwrite all attributes", i.e. _here's how I want my event to look like_, or all updates should be "additive" (as in patching), i.e. if I send metadata, don't replace, but add such that existing keys/values are left untouched (this is also an issue for any attached `labels`).

What I did with `AssetsAPI.create_hierachy` was to have an `upsert_mode` keyword argument that could be set to either replace or patch.

Reply:
At least the many use cases I know about, would like this to be either create this event if it does not exists, or overwrite it with this new information if it does.